/*
Copyright 2020 The Magma Authors.

This source code is licensed under the BSD-style license found in the
LICENSE file in the root directory of this source tree.

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.
*/

syntax = "proto3";

import "orc8r/protos/identity.proto";
import "google/protobuf/timestamp.proto";
import "google/protobuf/duration.proto";

package magma.orc8r;
option go_package = "magma/orc8r/lib/go/protos";

enum CertType {
    DEFAULT = 0; // Bootstrap cert
    VPN = 1;
}

message CSR {
    Identity id = 1;
    google.protobuf.Duration valid_time = 2;
    bytes csr_der = 3;// certificate signing request in DER encoding
    CertType cert_type = 4;
}

message Certificate {
    message SN {
        string sn = 1;
    }

    SN sn = 1;
    google.protobuf.Timestamp not_before = 2;
    google.protobuf.Timestamp not_after = 3;
    bytes cert_der = 4; // signed certificate in DER encoding
}

message CACert {
    bytes cert = 1; // ca certificate in DER encoding
}
